networking_cisco/plugins/cisco/cfg_agent/device_drivers/asr1k/aci_asr1k_cfg_syncer.py

```diff
@@ -0,0 +1,207 @@
+
+
+class ConfigSyncer(syncer.ConfigSyncer):
+
+    def __init__(self, router_db_info, driver,
+                 hosting_device_info, test_mode=False):
+        super(ConfigSyncer, self).__init__(router_db_info,
+                                           driver,
+                                           hosting_device_info,
+                                           test_mode=test_mode)
+
+    def delete_invalid_cfg(self, conn=None):
+        router_id_dict = self.router_id_dict
+        intf_segment_dict = self.intf_segment_dict
+        segment_nat_dict = self.segment_nat_dict
+
+        if not conn:
+            conn = self.driver._get_connection()
+
+        LOG.info(_LI("*************************"))
+        LOG.info(_LI("neutron router db records"))
+        for router_id, router in six.iteritems(router_id_dict):
+            LOG.info(_LI("ROUTER_ID: %s"), router_id)
+
+        LOG.info(_LI("\n"))
+
+        for segment_id, intf_list in six.iteritems(intf_segment_dict):
+            LOG.info(_LI("SEGMENT_ID: %s"), (segment_id))
+            for intf in intf_list:
+                dev_owner = intf['device_owner']
+                dev_id = intf['device_id'][0:6]
+                ip_addr = intf['fixed_ips'][0]['ip_address']
+                intf_info = "    INTF: %s, %s, %s" % \
+                            (ip_addr, dev_id, dev_owner)
+                LOG.info(intf_info)
+
+        running_cfg = self.get_running_config(conn)
+        parsed_cfg = HTParser(running_cfg)
+
+        invalid_cfg = []
+
+        invalid_cfg += self.clean_snat(conn,
+                                       router_id_dict,
+                                       intf_segment_dict,
+                                       segment_nat_dict,
+                                       parsed_cfg)
+
+        invalid_cfg += self.clean_nat_pool_overload(conn,
+                                                    router_id_dict,
+                                                    intf_segment_dict,
+                                                    segment_nat_dict,
+                                                    parsed_cfg)
+
+        invalid_cfg += self.clean_nat_pool(conn,
+                                           router_id_dict,
+                                           intf_segment_dict,
+                                           segment_nat_dict,
+                                           parsed_cfg)
+
+        if (cfg.CONF.multi_region.enable_multi_region):
+            set_route_regex = SET_ROUTE_MULTI_REGION_REGEX
+        else:
+            set_route_regex = SET_ROUTE_REGEX
+
+        invalid_cfg += self.clean_routes(conn,
+                                       router_id_dict,
+                                       intf_segment_dict,
+                                       segment_nat_dict,
+                                       parsed_cfg,
+                                       set_route_regex)
```

Comment:
minor: add a empty line here as it seems that's convention inside this function?